### PR TITLE
PE-17: Migration of provisiond XSD to DB

### PR DESCRIPTION
**Pull request by patrick-schweizer**
Jira: https://issues.opennms.org/browse/PE-17

### CODE REVIEW

File: features/config/service/api/src/main/java/org/opennms/features/config/service/api/ConfigurationManagerService.java [hunk 1]
```diff
@@ -60,6 +63,8 @@
     <ENTITY> void registerSchema(String configName, int majorVersion, int minorVersion,
                                  int patchVersion, Class<ENTITY> entityClass) throws IOException, JAXBException;
 
+    <ENTITY> void registerSchema(String configName, Version version, Class<ENTITY> entityClass) throws IOException, JAXBException;
```
Comment (patrick-schweizer) [inline features/config/service/api/src/main/java/org/opennms/features/config/service/api/ConfigurationManagerService.java:66]: I changed the interface slightly to group the version information into one object. @freddy61025 do you think this makes sense?

File: features/config/upgrade/src/test/java/org/opennms/config/upgrade/LiquibaseUpgraderIT.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                .withLogConsumer(new Slf4jLogConsumer(LOG));
+        container.start();
+        dataSource = createDatasource();
+    }
+
+    private static DataSource createDatasource() {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(String.format("jdbc:postgresql://localhost:%s/", container.getFirstMappedPort()));
+        config.setUsername("postgres");
+        config.setPassword("password");
```
Comment (freddy61025) [inline features/config/upgrade/src/test/java/org/opennms/config/upgrade/LiquibaseUpgraderIT.java:94]: Just curious why not use @JUnitTemporaryDatabase?

File: features/telemetry/protocols/openconfig/itests/src/test/java/org/opennms/features/telemetry/protocols/openconfig/OpenConfigIT.java [hunk 3]
```diff
@@ -74,9 +74,11 @@
 @ContextConfiguration(locations={
         "classpath:/META-INF/opennms/applicationContext-soa.xml",
         "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
         "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
         "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
         "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
```
Comment (freddy61025) [inline features/telemetry/protocols/openconfig/itests/src/test/java/org/opennms/features/telemetry/protocols/openconfig/OpenConfigIT.java:81]: It seems duplicated.

File: features/topology-map/plugins/org.opennms.features.topology.plugins.browsers/src/test/java/org/opennms/features/topology/plugins/browsers/OnmsDaoContainerIT.java [hunk 4]
```diff
@@ -57,9 +57,11 @@
 @ContextConfiguration(locations={
         "classpath:/META-INF/opennms/applicationContext-soa.xml",
         "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
         "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
         "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
         "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
         "classpath*:/META-INF/opennms/component-service.xml",
```
Comment (freddy61025) [inline features/topology-map/plugins/org.opennms.features.topology.plugins.browsers/src/test/java/org/opennms/features/topology/plugins/browsers/OnmsDaoContainerIT.java:65]: Duplicated

File: integrations/opennms-wsman-asset-provisioning-adapter/src/test/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapterIT.java [hunk 5]
```diff
@@ -62,7 +62,9 @@
         "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
         "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
         "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
         "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
```
Comment (freddy61025) [inline integrations/opennms-wsman-asset-provisioning-adapter/src/test/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapterIT.java:67]: duplicated

File: opennms-base-assembly/pom.xml [hunk 6]
```diff
@@ -445,6 +445,11 @@
       <groupId>org.opennms.features</groupId>
       <artifactId>opennms-integration-remedy</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.opennms.features.config.service</groupId>
+      <artifactId>org.opennms.features.config.service.impl</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```
Comment (freddy61025) [inline opennms-base-assembly/pom.xml:452]: It seems you do not rebase with CM-Phase1 yet.

File: opennms-dao/pom.xml [hunk 7]
```diff
@@ -53,6 +53,16 @@
...
+    <dependency>
+      <groupId>org.opennms.features.config</groupId>
+      <artifactId>org.opennms.features.config.upgrade</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.opennms.features.config.service</groupId>
+      <artifactId>org.opennms.features.config.service.api</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```
Comment (freddy61025) [inline opennms-dao/pom.xml:65]: org.opennms.features.config.service.api is included in CM-Phase1 already.

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v1/AcknowledgmentRestServiceIT.java [hunk 8]
```diff
@@ -71,6 +71,7 @@
         "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
         "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
         "classpath:/META-INF/opennms/applicationContext-dao.xml",
+		"classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml",
```
Comment (freddy61025) [inline opennms-webapp-rest/src/test/java/org/opennms/web/rest/v1/AcknowledgmentRestServiceIT.java:74]: There is 2 tab but not spaces.

**Review by freddy61025 — COMMENTED** [review]
Looks good to me. Just some minor issue and it seems not rebase with CM-Phase1 yet.

File: integrations/opennms-wsman-asset-provisioning-adapter/src/test/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapterIT.java (hunk 5 above)
Comment (patrick-schweizer) [reply integrations/opennms-wsman-asset-provisioning-adapter/src/test/java/org/opennms/netmgt/provision/WsManAssetProvisioningAdapterIT.java:67]: good catch! fixed.

File: features/topology-map/plugins/org.opennms.features.topology.plugins.browsers/src/test/java/org/opennms/features/topology/plugins/browsers/OnmsDaoContainerIT.java (hunk 4 above)
Comment (patrick-schweizer) [reply features/topology-map/plugins/org.opennms.features.topology.plugins.browsers/src/test/java/org/opennms/features/topology/plugins/browsers/OnmsDaoContainerIT.java:65]: good catch! fixed.

File: features/telemetry/protocols/openconfig/itests/src/test/java/org/opennms/features/telemetry/protocols/openconfig/OpenConfigIT.java (hunk 3 above)
Comment (patrick-schweizer) [reply features/telemetry/protocols/openconfig/itests/src/test/java/org/opennms/features/telemetry/protocols/openconfig/OpenConfigIT.java:81]: fixed

File: opennms-webapp-rest/src/test/java/org/opennms/web/rest/v1/AcknowledgmentRestServiceIT.java (hunk 8 above)
Comment (patrick-schweizer) [reply opennms-webapp-rest/src/test/java/org/opennms/web/rest/v1/AcknowledgmentRestServiceIT.java:74]: fixed.

File: opennms-dao/pom.xml (hunk 7 above)
Comment (patrick-schweizer) [reply opennms-dao/pom.xml:65]: I am not sure I understand?

**Pull request merged**